### PR TITLE
add symlink support in input blob

### LINKED ISSUE #31: Support symbolic links
**What would you like to be added**:
The `component-cli` should support symbolic links for the `CONTENT_PATH` feature.

**Why is this needed**:
I want to make use of the `CONTENT_PATH` feature and thus I have added my content under the `blueprint` directory.
However I have created a symbolic link (with name `config`) as I do not want to duplicate the content in my repository. When I run 

```bash
ls -la
total 8
drwxr-xr-x 5 me wheel 160 Aug 11 12:47 .
drwxr-xr-x 5 me wheel 160 Aug  5 21:18 ..
-rw-r--r-- 1 me wheel 963 Aug 10 16:41 blueprint.yaml
lrwxr-xr-x 1 me wheel  12 Aug 11 12:47 config -> ../../config
-rw-r--r-- 1 me wheel 496 Aug  9 14:17 gardenlogin-deploy-execution.yaml
```

```bash
component-cli component-archive resources add "${CA_PATH}" "${RESOURCES_FILE_PATH}"
```
I get the error message
```bash
info	add input blob from "./blueprint"
unable to tar input artifact: unable to open file "/config": openfile /config: AbsPath: too many links
make: *** [cnudie-generate-cd] Error 1
```

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:

Adds support symlinks in input blobs.
Symlinks are not automatically followed but only when configured to avoid security vulnerabilities where arbitrary files like secrets can be read.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/component-cli/issues/31

**Special notes for your reviewer**:

**Release note**:

```feature user
Input blobs defined in resources can now contain symlinks that are followed if configured.
See the resource input blob documentation for the syntax.
```

### CODE REVIEW

**Review by robertgraeff — APPROVED** [review]

File: pkg/commands/componentarchive/input/input.go [hunk 1]
```diff
@@ -214,74 +201,110 @@ func (input *BlobInput) Read(fs vfs.FileSystem, inputFilePath string) (*BlobOutp
...
+	case info.Mode().IsRegular():
 		if err := tw.WriteHeader(header); err != nil {
 			return fmt.Errorf("unable to write header for %q: %w", path, err)
 		}
-		if info.IsDir() {
-			return nil
-		}
-
-		file, err := fs.OpenFile(path, os.O_RDONLY, os.ModePerm)
+		file, err := fs.OpenFile(realPath, os.O_RDONLY, os.ModePerm)
```
Comment (achimweigel) [inline pkg/commands/componentarchive/input/input.go:307]: this was not changed in this PR but the file is not closed if an error occurs before in the cop function.

**Review by achimweigel — CHANGES_REQUESTED** [review]

File: pkg/commands/componentarchive/input/input.go (hunk 1 above)
Comment (schrodit) [reply pkg/commands/componentarchive/input/input.go:307]: done

**Review by achimweigel — APPROVED** [review]
/lgtm

**Pull request merged**